dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/source/utils/AuditUtils.java

```diff
@@ -81,62 +81,69 @@ public static String getSessionID(HttpServletRequest req) {
         final HttpServletRequest f_req = req;
 
         try {
-            sessionID = AccessController.doPrivileged(new PrivilegedExceptionAction<String>() {
-                @Override
-                public String run() throws Exception {
-                    HttpSession session = f_req.getSession();
-                    if (session != null) {
-                        return session.getId();
-                    } else {
-                        return null;
-                    }
-                }
-            });
-        } catch (PrivilegedActionException e) {
-            if ((e.getException()) instanceof com.ibm.websphere.servlet.session.UnauthorizedSessionRequestException) {
-                if (!req.isRequestedSessionIdFromCookie()) {
-                    sessionID = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                        @Override
-                        public String run() {
-                            return f_req.getSession().getId();
-                        }
-                    });
-                } else {
-                    sessionID = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                        @Override
-                        public String run() {
-                            return f_req.getRequestedSessionId();
-                        }
-                    });
-                }
+            sessionID = getSessionIDPrivileged(f_req);
+        } catch (PrivilegedActionException | com.ibm.websphere.servlet.session.UnauthorizedSessionRequestException e) {
```

Comment:
This changes the behavior.  Before we only did this logic if a PrivilegedActionException happened and the exception that is wrapped is a UnauthorizedSessionRequestException.

Reply:
I agree it will be a behavior change. I put back the original condition to keep the behavior.